### PR TITLE
Done add_coord_to_db

### PR DESCRIPTION
**Pull request by Alekseikovtun**

### CODE REVIEW

File: config/db_config.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import os
+
+from dotenv import load_dotenv
```
Comment (vdumchiviy) [inline config/db_config.py:3]: I prefer to control which file I need to read:
```from pathlib import Path
import environ
env = environ.Env()
# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent
environ.Env.read_env(BASE_DIR / ".env")

SECRET_KEY = env("SECRET_KEY")
```

File: models/station.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Base = declarative_base()
+
+db = f'postgresql://{db_config.POSTGRES_USER}:{db_config.POSTGRES_PASSWORD}@localhost:{db_config.POSTGRES_OUT_PORT}/postgres'
+engine = sa.create_engine(db)
+Session = sessionmaker(bind=engine)
+connection = engine.connect()
+
+
+class DronStatus(Base):
```
Comment (vdumchiviy) [inline models/station.py]: I assume that you need to separate files into station.py and dron.py

File: models/station.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship, sessionmaker
+
+import config.db_config as db_config
```
Comment (vdumchiviy) [inline models/station.py]: from config import db_config?

File: models/station.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Base = declarative_base()
+
+db = f'postgresql://{db_config.POSTGRES_USER}:{db_config.POSTGRES_PASSWORD}@localhost:{db_config.POSTGRES_OUT_PORT}/postgres'
```
Comment (vdumchiviy) [inline models/station.py]: The "models" folder should store only models. Move the initialization to a service folder.

File: models/station.py (hunk 2 above)
Comment (vdumchiviy) [inline models/station.py]: typo Drone, not Dron

File: models/station.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+class Task(Base):
+    __tablename__ = 'task'
+    # created_at = sa.Column()
+    # updated_at = sa.Column()
+    id = sa.Column(sa.Integer, primary_key=True)
+    latitude = sa.Column(sa.Integer)
+    longtitude = sa.Column(sa.Integer)
+
+    task_statuses = relationship('TaskStatus', backref='task')
```
Comment (vdumchiviy) [inline models/station.py]: I think you need to use a foreign key FROM Task TO TaskStatus, isn't it?

File: main.py [hunk 6]
```diff
@@ -1,47 +1,30 @@
...
-            coord1 = last_row[0]
-            coord2 = last_row[1]
-            print(create_json(coord1, coord2))
-        if txt == 'send':
-            add_new_row_to_csv(csv_file_path)
-        if txt == 'end':
+            # End-of-file error handler
+            pass
+        if txt == 'a':
+            db.add_coordinate_to_db()
```
Comment (vdumchiviy) [inline main.py]: You should input data here

File: service/db.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def read_data_for_new_task():
+    pass
+
+
+def add_coordinate_to_db():
+    last_coord = session.query(Task, func.max(Task.id)).group_by(Task.id)
+    last_id = last_coord.all()[0][1]
+    add_latitude = input('Enter the latitude coordinates: ')
```
Comment (vdumchiviy) [inline service/db.py]: service/db should responsible only for DB operations. Avoid entering data here.

**Review by vdumchiviy — CHANGES_REQUESTED** [review]

File: config/db_config.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import os
```
Comment (vdumchiviy) [inline config/db_config.py:1]: Choose "Discard changes" or add to gitignore that files

File: config/db_config.py (hunk 1 above)
Comment (vdumchiviy) [inline config/db_config.py:3]: try to change to environ

File: config/db_config.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+POSTGRES_OUT_PORT = os.getenv('POSTGRES_OUT_PORT')
+POSTGRES_USER = os.getenv('POSTGRES_USER')
+POSTGRES_PASSWORD = os.getenv('POSTGRES_PASSWORD')
```
Comment (vdumchiviy) [inline config/db_config.py]: where is the name of the database? If you change your RDBMS from Postgres to MySQL - you will not have DB named "postgres" at all.

File: models/drone.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+
+Base = declarative_base()
```
Comment (vdumchiviy) [inline models/drone.py:8]: Add a new file db_base_class.py in the model's folder:
```
from sqlalchemy_mixins import AllFeaturesMixin

class BaseModel(AllFeaturesMixin):
    __abstract__ = True
    deleted_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), default=func.now())
    updated_at: datetime = Column(
        DateTime(timezone=True), default=func.now(), onupdate=func.now()
    )

    __mapper_args__ = {"eager_defaults": True}

```

and in this file just use:
```
from db_base_class import BaseModel
from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, Numeric, String

class DroneStatus(BaseModel):
....
```

File: models/station.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+
+Base = declarative_base()
```
Comment (vdumchiviy) [inline models/station.py:8]: the same as for models/drone.py

File: service/db.py [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    next: Task = session.query(Task).filter(Task.id == next_task).all()[0]
+    result = {
+        'latitude': next.latitude,
+        'longitude': next.longitude
+    }
+    return result
+
+
+def add_coordinate_to_db(add_latitude, add_longitude):
+    last_coord = session.query(func.max(Task.id)).scalar()
```
Comment (vdumchiviy) [inline service/db.py]: last_coord_id ?

File: service/db.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+        session.commit()
+        print('\n Information about the drone has been received \n')
+        task_status_update = session.query(Task).filter(
+            Task.latitude == departure_lat,
+            Task.longitude == departure_long
+        )
+        record = task_status_update.one()
+        record.task_status_id = 3
+        session.commit()
+        print('\n The task status has been updated \n')
```
Comment (vdumchiviy) [inline service/db.py]: This is not an "add_dron_status", this is "update_task_info".

**Review by vdumchiviy — CHANGES_REQUESTED** [review]

File: models/drone.py (hunk 10 above)
Comment (Alekseikovtun) [reply models/drone.py:8]: Explicit initialization of the base class failed, but I added "created_at" and the changing "updated_at" columns to the Task and DroneStatus classes

**Pull request merged**